src/components/shareable-url-modal/shareable-url-modal.js

```diff
@@ -143,341 +176,93 @@ const ShareableUrlModal = ({ onToggleModal, visible }) => {
-                    value={value}
-                  />
-                ))}
-              </Dropdown>
-            </div>
-            <div className="shareable-url-modal__input-wrapper">
-              <div className="shareable-url-modal__input-label">
-                Bucket Name
-              </div>
-              <Input
-                defaultValue={bucket_name}
-                onChange={(value) => onChange('bucket_name', value)}
-                placeholder="Enter name"
-                resetValueTrigger={visible}
-                size="small"
-                type="input"
-                dataTest={'bucket_name'}
-              />
-            </div>
-            <div className="shareable-url-modal__input-wrapper">
-              <div className="shareable-url-modal__input-label">
-                Endpoint Link
-              </div>
-              <Input
-                defaultValue={endpoint}
-                onChange={(value) => onChange('endpoint', value)}
-                placeholder="Enter url"
-                resetValueTrigger={visible}
-                size="small"
-                type="input"
-                dataTest={'endpoint_name'}
-              />
-            </div>
-          </div>
-        </div>
-        <div className="shareable-url-modal__button-wrapper shareable-url-modal__button-wrapper--right">
-          <Button
-            mode="secondary"
-            onClick={() => handleModalClose()}
-            size="small"
-          >
-            Cancel
-          </Button>
-          <Button
-            disabled={!Object.values(isFormDirty).every((value) => value)}
-            size="small"
-            onClick={handleSubmit}
-          >
-            Publish
-          </Button>
-        </div>
-      </>
-    ) : null;
-  };
-
-  const { platform, bucket_name, endpoint } = inputValues || {};
+  const { platform } = inputValues || {};
 
   return (
     <Modal
       className={classnames('shareable-url-modal', {
         'shareable-url-modal__non-default-wrapper':
           deploymentState !== 'default',
+        'shareable-url-modal__published-wrapper':
+          deploymentState === 'published',
+        'shareable-url-modal__success-wrapper': deploymentState === 'success',
       })}
       closeModal={handleModalClose}
-      message={getDeploymentStateByType('message')}
-      title={getDeploymentStateByType('title')}
+      message={getDeploymentStateByType(
+        'message',
+        deploymentState,
+        compatibilityData,
+        shareableUrlMessages
+      )}
+      title={getDeploymentStateByType(
+        'title',
+        deploymentState,
+        compatibilityData,
+        shareableUrlMessages
+      )}
       visible={visible.shareableUrlModal}
     >
-      {renderCompatibilityMessage()}
-      {!isDisclaimerViewed && canUseShareableUrls ? (
-        renderDisclaimerContent()
+      {!isCompatible ? (
+        <CompatibilityView onClick={handleModalClose} />
```

Comment:
[Nit] This seems to be a view for compatibility error. Can we rename it to be CompatibilityErrorView ?

Reply:
yeah good suggestion, will do that